front/app/components/AssignBudgetControl/index.tsx

```diff
@@ -98,184 +87,181 @@ const StyledPBExpenses = styled(PBExpenses)`
 
 type TView = 'ideaCard' | 'ideaPage';
 
-interface OuterProps {
+interface Props {
   view: TView;
   projectId: string;
   ideaId: string;
   className?: string;
 }
 
-interface InnerProps extends OuterProps {
-  authUser: GetAuthUserChildProps;
-  idea: GetIdeaChildProps;
-  project: GetProjectChildProps;
-  phases: GetPhasesChildProps;
-  participationContext: IProjectData | IPhaseData;
-  participationContextId: string;
-  participationContextType: IParticipationContextType;
-}
-
 const AssignBudgetControl = memo(
-  ({
-    view,
-    ideaId,
-    authUser,
-    idea,
-    phases,
-    participationContext,
-    participationContextId,
-    participationContextType,
-    className,
-  }: InnerProps) => {
+  ({ view, ideaId, className, projectId }: Props) => {
+    const authUser = useAuthUser();
+    const idea = useIdea({ ideaId });
+    const project = useProject({ projectId });
+    const phases = usePhases(projectId);
+    const isContinuousProject =
+      project?.attributes.process_type === 'continuous';
+    const ideaPhaseIds = !isNilOrError(idea)
+      ? idea?.relationships?.phases?.data?.map((item) => item.id)
+      : null;
+    const ideaPhases = !isNilOrError(phases)
+      ? phases?.filter((phase) => includes(ideaPhaseIds, phase.id))
+      : null;
+    const latestRelevantIdeaPhase = ideaPhases
+      ? getLatestRelevantPhase(ideaPhases)
+      : null;
+    const participationContext = isContinuousProject
+      ? project
+      : latestRelevantIdeaPhase;
+    const participationContextType = isContinuousProject ? 'project' : 'phase';
+    const participationContextId = participationContext?.id || null;
```

Comment:
Deleted the 'wrapper' component and brought data loading and var creation inside the component to simplify it.